shared/js/background/classes/sw-tab.es6.js

```diff
@@ -0,0 +1,62 @@
+const Tab = require('./tab.es6')
+
+/**
+ * A stub tab implementation for service workers.
+ *
+ * For most actions, this just mimics a tab with the service worker URL as the tab URL. However, for cases
+ * such as tracker counts, it distributes stats to all known tabs with the same origin.
+ */
+class ServiceWorkerTab extends Tab {
+    /**
+     * @param {string} swUrl
+     * @param {Record<number, Tab>} tabContainer
+     */
+    constructor (swUrl, tabContainer) {
+        super({
+            tabId: -1,
+            url: swUrl,
+            status: null
+        })
+        this.origin = new URL(swUrl).origin
+        this.tabContainer = tabContainer
+    }
+
+    /**
+     * Find the list of tabs which share the same origin as this service worker.
+     * @returns {Tab[]}
+     */
+    _findMatchingTabs () {
+        return Object.keys(this.tabContainer).filter(tabId => {
+            const tab = this.tabContainer[tabId]
+            try {
+                return Number(tabId) > -1 && new URL(tab.url).origin === this.origin
```

Comment:
Why cast the tabId to a number? If it's not a number, what value will that result in?

Reply:
This is due to type checking - because object keys are strings, then `tabId` here is a string. Using `Number` casts it to a number for the comparison. If it's not a number, `Number(tabId) === NaN`, and `NaN > -1` is `false`, so I think we're good here.

Reply:
OK cool.

Reply:
(So I guess this is just another way of writing `parseInt(tabId, 10)`?)